core/trino-spi/pom.xml

```diff
@@ -35,6 +35,11 @@
             <optional>true</optional>
```

Comment:
> All annotated elements are non-public,

After a second look, I think we should abandon this PR. It is not going to any plugin implementation and as you have already noticed (with `There are no mis-uses`) we are solving a problem that do not manifest today. So I think the price of adding new dependency to SPI is too high comparing to benefits.